src/ParallelTempering.jl

```diff
@@ -90,16 +104,56 @@ function replica_exchange!(replica::Replica)
     end
 
     # accept replica exch.: swap configurations
-    MPI.Sendrecv!(replica.sampler.system.sites, rex_rank, 3,
-                  replica.sampler.system.sites, rex_rank, 3, MPI.COMM_WORLD)
-
+    MPI.Sendrecv!(        
+        deepcopy(replica.sampler.system.sites), rex_rank, 3,
+                 replica.sampler.system.sites , rex_rank, 3, 
+        MPI.COMM_WORLD
+    )
     # recalculate energy and magnetization of new state
     replica.sampler.E = energy(replica.sampler.system)
     replica.sampler.M = sum(replica.sampler.system)
```

Comment:
See comments below (noticed this one late). Please use `reset_running_energy!` and `reset_running_mag!`.